app/controllers/guest_device_controller.rb

```diff
@@ -0,0 +1,50 @@
+class GuestDeviceController < ApplicationController
+  include Mixins::GenericListMixin
+  include Mixins::GenericShowMixin
+  include Mixins::GenericSessionMixin
+  include Mixins::MoreShowActions
+
+  before_action :check_privileges
+  before_action :session_data
+  after_action :cleanup_action
+  after_action :set_session_data
+
+  def self.model
+    @model ||= "GuestDevice".safe_constantize
+  end
+
+  def title
+    _('Guest Devices')
+  end
+
+  def model
+    self.class.model
+  end
+
+  def self.table_name
+    @table_name ||= "guest_devices"
+  end
+
+  def session_data
+    @title  = _("Guest Devices")
+    @layout = "guest_device"
+    @lastaction = session[:guest_device_lastaction]
+  end
+
+  def set_session_data
+    session[:layout] = @layout
+    session[:guest_device_lastaction] = @lastaction
+  end
+
+  def show_list
+    options = {:model => "GuestDevice", :named_scope => [:with_ethernet_type]}
+    process_show_list(options)
+  end
```

Comment:
@skovic I think you can remove the `show_list` override from here since we are not using it anymore.

Reply:
@AparnaKarve I removed the override for show_list, and named_scope still seems to work. How is it still working without this override? Just curious. Thanks

Reply:
The nested display list is using `named_scope`, and that's all we need to make `named_scope` work.

Note that the nested display list is an action that is called on `physical_server_controller` (the parent controller )and not on `guest_device_controller`.
In other words, `GuestDeviceController#show_list` is not being used at all -- hence you can remove that method from the controller and even `routes.rb`.

Reply:
@AparnaKarve OK, got it. Thank you for the explanation. I'll remove the override.